lib/beautiful/init.lua

```diff
@@ -117,14 +117,25 @@ function beautiful.init(config)
         if type(config) == 'string' then
             -- Expand the '~' $HOME shortcut
             config = config:gsub("^~/", homedir .. "/")
+            configdir = config:gsub("/[^/]+$", "")
```

Comment:
Just for completeness, something like `sh -c 'cd /usr/local/etc && /usr/local/bin/awesome -c awesome/rc.lua` would have the issue I meant, because here the path to the config is a relative path. However, as I wrote below, the code already assumes absolute paths elsewhere and thus we can ignore this here.